src/main/java/io/supertokens/webserver/api/emailpassword/UserAPI.java

```diff
@@ -137,7 +137,7 @@ protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IO
             if (userIdMapping != null) {
                 userId = userIdMapping.superTokensUserId;
             }
-            EmailPassword.updateUsersEmailOrPassword(this.getTenantIdentifier(req), main, userId, email, password);
```

Comment:
user ID mapping that's happening above will also have to find the right storage layer wherein the user exists.